root/static/scripts/edit/URLCleanup.js

```diff
@@ -2910,7 +2910,42 @@ const CLEANUPS: CleanupEntries = {
   },
   'jaxsta': {
     match: [new RegExp('^(https?://)?(www\\.)?jaxsta\\.com', 'i')],
-    restrict: [LINK_TYPES.otherdatabases],
+    restrict: [
+      LINK_TYPES.otherdatabases,
+      {work: [LINK_TYPES.otherdatabases.work, LINK_TYPES.lyrics.work]},
+    ],
+    select: function (url, sourceType) {
+      const m = /^https:\/\/jaxsta\.com\/(\w+)\/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}(?:\/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})?$/.exec(url);
+      if (m) {
+        const prefix = m[1];
+        switch (prefix) {
+          case 'profile':
+            if (sourceType === 'artist') {
+              return LINK_TYPES.otherdatabases.artist;
+            }
+            if (sourceType === 'label') {
+              return LINK_TYPES.otherdatabases.label;
+            }
+            break;
+          case 'recording':
+            if (sourceType === 'recording') {
+              return LINK_TYPES.otherdatabases.recording;
+            }
+            break;
+          case 'release':
+            if (sourceType === 'release') {
+              return LINK_TYPES.otherdatabases.release;
+            }
+            break;
```

Comment:
This is not needed (but for `work`) because there is only one possible relationship type (through `restrict`) for each of these entity types.

Reply:
Hmm, ok. If that's the case we should also drop the extra stuff for `mainlynorfolk` then, that's what I was following :)

Reply:
Yes.